src/Features/Core/Portable/Completion/IFeaturesCustomCommitCompletionProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    /// <summary>
+    /// The ICustomCommitCompletionProvider in the EditorFeatures layer marks completion providers
+    /// that do fancy things with the caret. In the move to Editor Completion, we use this
+    /// interface to mark <see cref="CompletionProvider" />s that need to do more than insert the
+    /// InsertText.
+    /// </summary>
+    interface IFeaturesCustomCommitCompletionProvider
```

Comment:
> My understanding that ICustomCommitCompletionProvider is obsolete

But... isn't this just replacing that obsolete interface with the same thing...?

Reply:
Yes, this is a similar thing. ICustomCommitCompletionProvider is used for some provider and this one for some other ones.

---
In reply to: [222090411](https://github.com/dotnet/roslyn/pull/29016#discussion_r222090411) [](ancestors = 222090411)

Reply:
Why can't we use one interface for both cases?

Reply:
1. IFeaturesCustomCommitCompletionProvider is defined in Microsoft.CodeAnalysis.Features and has no references to 
using Microsoft.VisualStudio.Text;
using Microsoft.VisualStudio.Text.Editor;

2. ICustomCommitCompletionProvider is defined in Microsoft.CodeAnalysis.EditorFeatures and uses
using Microsoft.VisualStudio.Text;
using Microsoft.VisualStudio.Text.Editor;

ICustomCommitCompletionProvider should be removed with the legacy completion removal. There is no reason to move IFeaturesCustomCommitCompletionProvider (it is used out of Microsoft.CodeAnalysis.EditorFeatures as well)
or add references to editor to Microsoft.CodeAnalysis.Features 


---
In reply to: [222129074](https://github.com/dotnet/roslyn/pull/29016#discussion_r222129074) [](ancestors = 222129074)

Reply:
ah, that makes more sense now.  thanks for clarifying that!

Reply:
Actually, i'm more confused having read ths review.  Some of our simplest intellisense cases are not IFeaturesCustomCommitCompletionProvider's.  Someting seems *super* wrong for that to be the case.  

Indeed, as far as i can tell IFeaturesCustomCommitCompletionProvider  is simply the GetChangeMethod that already exists in CompletionService.  in which case, 100% of our providers/services should implement this.  

So i don't see the purpose of this interface at all.  We should *always* be deferrin to teh CompletionService to figure out what change ot make.  it will do the right thing, and there's no reason i can tell to ever do otehrwise..